json/src/main/scala/fs2/data/json/tokens.scala

```diff
@@ -55,7 +62,298 @@ object Token {
     def jsonRepr: String = value
   }
   case class StringValue(value: String) extends Token("string") {
-    def jsonRepr: String = s""""$value""""
+    def jsonRepr: String = {
+      val rendered = new StringBuilder
+      rendered.append('"')
+      renderString(value, 0, rendered)
+      rendered.append('"')
+      rendered.result()
+    }
+  }
+
+  private final val hex = "0123456789abcdef"
```

Comment:
```suggestion
  private final val hex = "0123456789abcdef".toArray
```

I'd expect an array of chars to be faster in index-based access (haven't benchmarked though).

Reply:
True, the string random access has some overhead.